internal/core/processor/universal/processor_test.go

```diff
@@ -167,121 +167,133 @@ func TestGetLastBlock(t *testing.T) {
 	}
 }
 
-func TestVerifyChainConsistency(t *testing.T) {
+func TestProcessUniversalBlockRange(t *testing.T) {
 	t.Parallel()
+
+	ctx := context.TODO()
 	tests := []struct {
-		name                   string
-		startingBlock          uint64
-		lastBlockDB            model.Block
-		lastBlockDBError       error
-		previousBlockData      *types.Header
-		previousBlockDataError error
-		expectedError          error
+		name                       string
+		startingBlock              uint64
+		previousBlockHeader        *types.Header
+		previousBlockData          model.Block
+		blockHeader                *types.Header
+		blockData                  model.Block
+		discoverReturn             bool
+		updateReturn               map[string][]model.ERC721Transfer
+		expectedError              error
+		expectedTxCommit           int
```

Comment:
Can `commit `happen if non `nil` is expected? Maybe I am wrong, but I think it is not possible.

Reply:
you mean non nil error? Yes, but only if the error happens in the commit. But I prefer to have two different variables for clarity.